internal/arduino/cores/cores.go

```diff
@@ -70,13 +71,59 @@ type PlatformRelease struct {
 	Programmers             map[string]*Programmer        `json:"-"`
 	Menus                   *properties.Map               `json:"-"`
 	InstallDir              *paths.Path                   `json:"-"`
+	Timestamps              *TimestampsStore              // Contains the timestamps of the files used to build this PlatformRelease
 	IsTrusted               bool                          `json:"-"`
 	PluggableDiscoveryAware bool                          `json:"-"` // true if the Platform supports pluggable discovery (no compatibility layer required)
 	Monitors                map[string]*MonitorDependency `json:"-"`
 	MonitorsDevRecipes      map[string]string             `json:"-"`
 	Compatible              bool                          `json:"-"` // true if at all ToolDependencies are available for the current OS/ARCH.
 }
 
+// TimestampsStore is a generic structure to store timestamps
+type TimestampsStore struct {
+	timestamps map[*paths.Path]*time.Time
+}
+
+// NewTimestampsStore creates a new TimestampsStore
+func NewTimestampsStore() *TimestampsStore {
+	return &TimestampsStore{
+		timestamps: map[*paths.Path]*time.Time{},
+	}
+}
+
+// AddFile adds a file to the TimestampsStore
+func (t *TimestampsStore) AddFile(path *paths.Path) {
```

Comment:
As it is now, the function will panic with a nil pointer exception, I think it's a more desirable behavior than returning an error because it means that we got something wrong...